gem/ci/build.ps1

```diff
@@ -118,39 +123,48 @@ function Clear-ContainerBuilds(
         Remove-Item $ENV:VOLUME_ROOT -Force -Recurse -ErrorAction Continue
     }
 
-    # this provides example data which ConvertFrom-String infers parsing structure with
-    $template = @'
+    Write-Host "`nPruning Volumes"
+    docker volume prune
+
+    Write-Host "`nPruning Networks"
+    docker network prune
+
+    if ($Name -ne $null)
+    {
+        # this provides example data which ConvertFrom-String infers parsing structure with
+        $template = @'
 {Version*:10.2.3*} {ID:5b84704c1d01} {[DateTime]Created:2019-02-07 18:24:51} +0000 GMT
 {Version*:latest} {ID:0123456789ab} {[DateTime]Created:2019-01-29 00:05:33} +0000 GMT
 '@
-    $output = docker images --filter=reference="$Namespace/${Name}" --format "{{.Tag}} {{.ID}} {{.CreatedAt}}"
-    Write-Output @"
+        $output = docker images --filter=reference="$Namespace/${Name}" --format "{{.Tag}} {{.ID}} {{.CreatedAt}}"
+        Write-Host @"
 
 Found $Namespace/${Name} images:
 $($output | Out-String)
 
 "@
 
-    if ($output -eq $null) { return }
-
-    Write-Output "Filtering removal candidates..."
-    # docker image prune supports filter until= but not repository like 'puppetlabs/foo'
-    # must use label= style filtering which is a bit more inconvenient
-    # that output is also not user-friendly!
-    # engine doesn't maintain "last used" or "last pulled" metadata, which would be more useful
-    # https://github.com/moby/moby/issues/4237
-    $output |
-      ConvertFrom-String -TemplateContent $template |
-      ? { $_.Created -lt $OlderThan } |
-      # ensure 'latest' are listed first
-      Sort-Object -Property Version -Descending |
-      % {
-        Write-Output "Removing Old $Namespace/${Name} Image $($_.Version) ($($_.ID)) Created On $($_.Created)"
-        $forcecli = if ($Force) { '-f' } else { '' }
-        docker image rm $_.ID $forcecli
-      }
-
-    Write-Output "`nPruning Dangling Images"
+        if ($output -eq $null) { return }
+
+        Write-Host 'Filtering removal candidates...'
+        # docker image prune supports filter until= but not repository like 'puppetlabs/foo'
+        # must use label= style filtering which is a bit more inconvenient
+        # that output is also not user-friendly!
+        # engine doesn't maintain "last used" or "last pulled" metadata, which would be more useful
+        # https://github.com/moby/moby/issues/4237
+        $output |
+          ConvertFrom-String -TemplateContent $template |
+          ? { $_.Created -lt $OlderThan } |
+          # ensure 'latest' are listed first
+          Sort-Object -Property Version -Descending |
+          % {
+            Write-Host "Removing Old $Namespace/${Name} Image $($_.Version) ($($_.ID)) Created On $($_.Created)"
+            $forcecli = if ($Force) { '-f' } else { '' }
+            docker image rm $_.ID $forcecli
+          }
+    }
+
+    Write-Host "`nPruning Dangling Images"
     docker image prune --filter "dangling=true" --force
```

Comment:
....like this, should it be moved up?

Reply:
This was also always here... because if we're not calling `docker image rm`, there shouldn't be dangling containers to cleanup.

Reply:
True, but it doesn't fail to to run it when there are no images, and it'll catch unexpected things like a failed cleanup from a previous run, or some leftovers from manual experimentation on the box 🤷‍♂

Reply:
I refactored it subtly into 4 separate operations in https://github.com/puppetlabs/pupperware/pull/98/commits/9e34288f9665bf8eecfeaf912f49877a96b54472

Dangling image cleanup will always happen now